### PR TITLE
Castle window animation fixes

### LINKED ISSUE #544: Acceleration of city animation while new building appears
During the time, new building appears, all animations in city accelerate significantly.

### LINKED ISSUE #585: No water animation of waterfall in Warlock city
In the original game water of waterfall is animated. But in our version we have animated only a bottom part of waterfall...
Waterfall

### LINKED ISSUE #870: Castle moat water animation is missing
Moat water animation is missing in all castles except Barbarian and half of Warlock castle moat.

### PR DESCRIPTION
**Pull request by idshibanov**
Fixes #544. Fixes #585. Fixes #870.

- Separate frame counter for construction animation.
- Adding color cycling for castle buildings.
- Increasing animation delay to make it smoother.
- Removing 4 unneeded colors from color cycle palette.

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @idshibanov , animation acceleration still happens during building construction. It's very noticeable in Sorceress castle for Tavern animation when a light in window flashes faster.

**Comment by idshibanov** [comment]
> Hi @idshibanov , animation acceleration still happens during building construction. It's very noticeable in Sorceress castle for Tavern animation when a light in window flashes faster.

Okay, found the source of the issue, will stay at the same frame. Don't know if there's a quick way to refactor the code to allow build and generic animations at the same time, that will be probably a rewrite.

File: src/fheroes2/agg/pal.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ namespace PAL
     {
         static std::vector<CyclingColorSet> cycleSet;
         if ( cycleSet.empty() ) {
-            const CyclingColorSet cycleData[] = {{0xD6, 4, false}, {0xDA, 4, false}, {0xE7, 5, true}, {0xEE, 4, false}, {0xF2, 4, false}};
```
[the remark this answers is not in the document]
Comment (idshibanov) [reply src/fheroes2/agg/pal.cpp:160]: It actually adds more flickering where it's not supposed to be (Wizard's castle for example or Warlock's moat). Original code is convoluted, I saw the latter 0xF2 color block swapping there but now I believe they don't overwrite those in final palette when running memcpy.

I tested different animations after the change (map, teleporters, elementals in battle); they all seem correct to me.

**Comment by ihhub** [comment]
Excellent work, @idshibanov !

**Pull request merged**